src/Service/NotificationReceiverService.php

```diff
@@ -250,16 +251,17 @@ private function isAuthorized($isTestNotification, $requestUser, $requestPasswor
      * Save notification into the database for cron job to execute notification
      *
      * @param $notificationItem
+     * @param $salesChannelId
      * @return bool
      * @throws AdyenException
      * @throws AuthenticationException
      * @throws HMACKeyValidationException
      * @throws MerchantAccountCodeException
      */
-    protected function processNotificationItem($notificationItem)
+    protected function processNotificationItem($notificationItem, $salesChannelId)
     {
         $merchantAccount = $this->configurationService->getMerchantAccount();
-        $hmacKey = $this->configurationService->getHmacKey();
+        $hmacKey = $this->configurationService->getHmacKey($salesChannelId);
```

Comment:
If the sales channel ID is empty I think it fetches the default value ("All Sales Channel"), if not then it will always get the Headless sales channel ID, can you confirm that?

Maybe setting the notifications HMAC in Headless is the way to go, let's talk about it later

Reply:
Yesterday I think we came to the conclusion to ask the merchants (via the documentation) to only set the HMAC for the default store configuration and only use this config from there and not for each store separately. Is this still needed here?

Reply:
We built the notification endpoint ((see here)[https://github.com/Adyen/adyen-shopware6/blob/develop/src/Storefront/Controller/NotificationReceiverController.php#L55]) to use the `storefront`scope and, from my investigation, the notification request comes in via the `storefront` salesChannelID and that is what will be used here to fetch the HMAC key.
Moreover, I think it is still needed because in the (`getHmacKey` function)[https://github.com/Adyen/adyen-shopware6/blob/develop/src/Service/ConfigurationService.php#L164] we still use the sales channel to query for the Hmac key, not passing the sales channel will return a php error.

Reply:
got it :)